tests/PHPStan/Analyser/nsrt/bug-11716.php

```diff
@@ -141,9 +133,46 @@ function emptyString($mixed)
 {
     // see https://3v4l.org/XHZdr
     $arr = ['' => 1, 'a' => 2];
+    if (isset($arr[$mixed])) {
+        assertType("0.0|''|'a'|null", $mixed); // could be ''|'a'|null
+    } else {
+        assertType('mixed', $mixed);
+    }
+    assertType('mixed', $mixed);
+}
+
+function numericString($mixed)
+{
+    $arr = ['1' => 1, '2' => 2];
+    if (isset($arr[$mixed])) {
+        assertType("1|1.0|2|2.0|'1'|'2'|true", $mixed);
```

Comment:
https://3v4l.org/Rml9s - we need https://github.com/phpstan/phpstan/issues/6963 or sadly `float` here